### PR TITLE
DriveFS: Handle receiving a stream instead of a node

### PR DESCRIPTION
**Pull request by martinRenou**
This will allow removing the monkey patch in https://github.com/jupyterlite/xeus/blob/main/src/worker.ts#L19

This monkey patch was introduced in https://github.com/jupyterlite/xeus-python-kernel/pull/27

### CODE REVIEW

File: packages/contents/src/emscripten.ts [hunk 1]
```diff
@@ -57,6 +57,12 @@ export interface IEmscriptenStream {
...
 }
 
+export function instanceOfStream(
+  nodeOrStream: IEmscriptenFSNode | IEmscriptenStream,
+): nodeOrStream is IEmscriptenStream {
+  return 'node' in nodeOrStream;
+}
+
 export interface IEmscriptenNodeOps {
   getattr(node: IEmscriptenFSNode): IStats;
```
Comment (jtpio) [inline packages/contents/src/emscripten.ts:67]: Should this be updated according to the changes above?

```suggestion
  getattr(node: IEmscriptenFSNode | IEmscriptenStream): IStats;
```

Comment (jtpio) [reply packages/contents/src/emscripten.ts:67]: If so, probably also for the other methods.

Comment (martinRenou) [reply packages/contents/src/emscripten.ts:67]: Ah right I missed it, thanks!

File: packages/contents/src/emscripten.ts [hunk 2]
```diff
@@ -57,22 +57,39 @@ export interface IEmscriptenStream {
...
+  nodeOrStream: IEmscriptenFSNode | IEmscriptenStream,
+): nodeOrStream is IEmscriptenStream {
+  return 'node' in nodeOrStream;
+}
+
 export interface IEmscriptenNodeOps {
-  getattr(node: IEmscriptenFSNode): IStats;
-  setattr(node: IEmscriptenFSNode, attr: IStats): void;
-  lookup(parent: IEmscriptenFSNode, name: string): IEmscriptenFSNode;
+  getattr(node: IEmscriptenFSNode | IEmscriptenStream): IStats;
```
Comment (jtpio) [inline packages/contents/src/emscripten.ts:67]: Looks like this change of signature can be considered as a breaking change?

Although it's probably not directly used that much downstream, should we still document it in the migration guide? https://jupyterlite.readthedocs.io/en/latest/migration.html

**Comment by martinRenou** [comment]
(adding back the enhancement label for the CI check)

**Pull request merged**